apps/query-eval/queryeval/driver.py

```diff
@@ -371,11 +400,38 @@ def do_eval(self, query: QueryEvalQuery, result: QueryEvalResult) -> QueryEvalRe
                 console.print("[red]:x: Documents retrieved don't include all expected docs")
                 console.print(f"Missing docs: {expected_doc_set - retrieved_doc_set})")
             metrics.doc_retrieval_recall = len(retrieved_doc_set & expected_doc_set) / len(expected_doc_set)
-            metrics.doc_retrieval_precision = len(retrieved_doc_set & expected_doc_set) / len(result.retrieved_docs)
+            metrics.doc_retrieval_precision = len(retrieved_doc_set & expected_doc_set) / len(retrieved_doc_set)
 
-        # Evaluate result
-        if not result.result:
-            console.print("[yellow] No query execution result available, skipping..", style="italic")
+        # Evaluate string metrics
+        if not query.expected:
+            console.print("[yellow]:construction: No expected response found, skipping.. ")
+        elif not result.result:
+            console.print(
+                "[yellow] No query execution result available, skipping..",
+                style="italic",
+            )
+        else:
+            if isinstance(query.expected, str) and isinstance(result.result, str):
+                sample = SingleTurnSample(
+                    response=result.result,
+                    reference=query.expected,
+                )
+                scores = asyncio.run(
```

Comment:
do we really need to make compute_string_metrics an async method? Can we just have it be sync and have it wait for processing within? 

This is adding a little complexity and I'm unsure why

Reply:
I looked into this as well for a long time. Ragas method to get score is async because of which we have to do this. Do you have any suggestions?

Reply:
You can do a variant off https://chatgpt.com/share/67466a08-1488-8007-b4fa-2d2538c74071

Basically do the waiting for async completion inside the compute_string_metrics method

Reply:
There will be no waiting as such because do_eval runs one sample at a time. Ragas people have made it an async function to handle large workloads.

Reply:
Regardless I have made the changes.

Reply:
Right I just mean in terms of function APIs